src/time-slot-bar-week/time-slot-bar-week-days.tsx

```diff
@@ -0,0 +1,510 @@
+                                    $variant={variant}
+                                    {...styleLabelProps}
+                                >
+                                    {day.format("D")}
+                                </DayLabelWeek>
+                            </InteractiveCircle>
+                        </GrowDayCell>
+                    );
+                })}
+            </HeaderCellWeekColumn>
+        );
+    };
+
+    const renderTimeColumn = () => {
+        let hasSetFirstAfternoonHour = false;
+
+        const formatTime = (index: number) => {
+            const parsedTime = dayjs(minStartTime, "HH:mm").add(
+                index * 4 * interval,
+                "minutes"
+            );
+            const hour = parsedTime.format("h");
+            const minutes = parsedTime.format("mm");
+            const amPm = parsedTime.format("a");
+
+            let formattedTime = `${hour}${
+                minutes !== "00" ? ` ${minutes}` : ""
+            }`;
+
+            if (
+                !hasSetFirstAfternoonHour &&
+                amPm === "pm" &&
+                minutes === "00"
+            ) {
+                formattedTime += ` ${amPm}`;
+                hasSetFirstAfternoonHour = true;
+            }
+
+            return formattedTime;
+        };
+
+        return (
+            <TimeColumn $height={height}>
+                {Array.from(
+                    { length: Math.ceil(numberOfCells / 4) },
+                    (_, index) => (
+                        <TimeColumnWrapper key={`time-${index}`}>
+                            <TimeColumnText>{formatTime(index)}</TimeColumnText>
+                        </TimeColumnWrapper>
+                    )
+                )}
+            </TimeColumn>
+        );
+    };
+
+    const renderTimeSlotBarCells = () => {
+        return (
+            <Expandable style={expandableStyles}>
+                <ColumnWeekCell
+                    ref={cellsRef}
+                    key={`week-cell-${calendarDate.format(dateFormat)}`}
+                >
+                    {currentCalendarWeek.map((day, dayIndex) => {
+                        const formattedDate = day.format(dateFormat);
+                        const cellsArray =
+                            generatedDaySlots[formattedDate] ??
+                            Array.from(
+                                {
+                                    length:
+                                        variant === "flexible"
+                                            ? numberOfCells
+                                            : 1,
+                                },
+                                (_, index) => generateFallbackCell(index)
+                            );
+                        return (
+                            <TimeSlotWrapper key={`wrapper-${dayIndex}`}>
+                                {cellsArray &&
+                                    cellsArray.map((slot) => {
```

Comment:
seems like cellsArray should always be initialised

```suggestion
                                {cellsArray.map((slot) => {
```